indexer/test/indexer.test.js

```diff
@@ -905,6 +954,28 @@ describe('POST /service-provider-registry/provider-removed', () => {
   })
 })
 
+async function withDataSet(
+  env,
+  { dataSetId = randomId(), withCDN = true, payer, payee },
+) {
+  await env.DB.prepare(
+    `
+    INSERT INTO data_sets (
+      id,
+      with_cdn,
+      payer,
+      payee
+    )
+    VALUES (?, ?, ?, ?)
+    ON CONFLICT DO NOTHING
```

Comment:
This is risky; it can hide test failure causes, making it difficult to troubleshoot. It can also make a test to always pass because the test data created by the test was discarded, and slightly different data from previous tests was used instead.

I propose to:
- Remove this `ON CONFLICT` line, let the function throw an error on conflicts
- Add `beforeEach` hooks to reset the database before each test to avoid conflicts.